src/Optimization/model_internal.jl

```diff
@@ -0,0 +1,94 @@
+mutable struct ModelInternal{T <: AbstractOptimizationContainer}
+    container::T
+    ic_model_container::Union{Nothing, T}
```

Comment:
`initial_conditions_model_container`? Or can this be something more generic? It seems to be an optional, secondary model container.

Reply:
It is common to many models to have initial conditions.